test/configuration_test.dart

```diff
@@ -201,7 +215,14 @@ void main() {
     );
     expect(
       uri1.toString(),
-      'http://world.openfoodfacts.org/test/test.pl?test=true&queryType=PROD',
+      'http://world.openfoodfacts.org/test/test.pl?test=true&queryType=PROD&$_appNameValue',
     );
   });
 }
+
+String get _appNameValue => 'app_name=${Uri.encodeFull(
+      OpenFoodAPIConfiguration.userAgent!.name.replaceAll(
+        ' ',
+        '+',
+      ),
+    )}';
```

Comment:
```suggestion
String get _appNameValue => 'app_name=${Uri.encodeComponent(
      OpenFoodAPIConfiguration.userAgent!.name,
    )}';
```

Reply:
Unfortunately it won't work, because the url encoder, transform spaces in `%20`, whereas we need `+` here.

Reply:
@g123k Understood. Still, using `encodeComponent` instead of `encodeFull` is more readable.